File: testdata/bench/javascript/BenchmarkTest00045.js
// Vulnerable: Command injection via execSync
const express = require('express');
const { execSync } = require('child_process');
const router = express.Router();
router.get('/test045', (req, res) => {
    const dir = req.query.dir;
    const result = execSync('ls -la ' + dir);
    res.send('<pre>' + result.toString() + '</pre>');
});
module.exports = router;

File: testdata/bench/kotlin/BenchmarkTest00045.kt
package bench

import io.ktor.server.application.*
import io.ktor.server.response.*

// Vulnerable: command injection via query parameter in ProcessBuilder
suspend fun handler00045(call: ApplicationCall) {
    val dir = call.request.queryParameters["dir"]
    val pb = ProcessBuilder("/bin/sh", "-c", "ls -la $dir")
    val process = pb.start()
    val output = process.inputStream.bufferedReader().readText()
    call.respondText(output)
}
